third_party/amd/backend/driver.py

```diff
@@ -124,8 +124,18 @@ def _get_path_to_hip_runtime_dylib():
     raise RuntimeError(f"cannot locate {lib_name} after attempted paths {paths}")
 
 
+@functools.lru_cache()
+def extra_cache_key():
+    # we need platform info (x86_64, aarch64 etc) and also HIP runtime info
```

Comment:
Ah that's interesting to know. Did you do deep investigation and can share more with us which is breaking between hip versions? Wanting to make sure we are not falling into similar issues in the future.

Reply:
The error is this: 

```
  File triton/backends/amd/driver.py", line 291, in __init__
    self.utils = HIPUtils()
  File "triton/backends/amd/driver.py", line 44, in __init__
    mod = compile_module_from_src(Path(os.path.join(dirname, "driver.c")).read_text(), "hip_utils")
  File "triton/backends/amd/driver.py", line 31, in compile_module_from_src
    mod = importlib.util.module_from_spec(spec)
  File "<frozen importlib._bootstrap>", line 730, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1176, in create_module
  File "<frozen importlib._bootstrap>", line 400, in _call_with_frames_removed
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
ImportError: /tmp/torchinductor_xdwang/triton/0/69ed2ebf5a90924d2ea00c0358d818ce7a4cc4ada919d4fd9bdb94650a0e67d6/hip_utils.so: undefined symbol: hipGetDevicePropertiesR0600```

I think this R0600 is from the API versioning of HIP runtime

Reply:
Oh yeah the `hipGetDevicePropertiesR0600` is a known issue. Sorry about it. Though we dynamically load HIP runtime now and I don't think we are using that symbol? https://github.com/triton-lang/triton/blob/main/third_party/amd/backend/driver.c#L22-L38. Is this producible at the `main` tip?